### ISSUE TITLE
Potential Problem assigning B.1.623 and B.1.621

### ISSUE THREAD

**Issue by luenling** [cov-lineages/pangolin#253]
Dear Pangolin team,

When assigning lineages to patient samples with pangolin I stumbled over a cluster that looks like B.1.621 but was assigned B.1.623.
The mutations we find definitely fit the B.1.621 in GISAID and not B.1.623, so either the lineage definition was changed or something else went wrong. I uploaded the sequences to USHER (https://hgwdev-angie.gi.ucsc.edu/cgi-bin/hgPhyloPlace) and got a strange result there too. Usher classified the sequence as B.1.621, but the closest sequences where B.1.623 (I guess classified by pangolin in GISAID).
An example of the AA changes our sequences have :
ORF1ab:T1055A; ORF1ab:T1538I; ORF1ab:N2147D; ORF1ab:T2952I; ORF1ab:T3255I; ORF1ab:Q3729R; ORF1ab:P4715L; ORF1ab:P5743S; ORF1ab:P5815S; ORF1ab:A6532V; ORF1ab:S6713L; S:T95I; S:Y144S; S:R346K; S:E484K; S:N501Y; S:D614G; S:P681H; S:D950N; ORF3a:Q57H; ORF3a:V256fs; ORF3a:V259L; ORF8:T11K; ORF8:P38S; ORF8:S67F; N:T205I

Of these the following fit B.1.621 in https://outbreak.info/situation-reports?pango=B.1.621
ORF1ab:T1055A; ORF1ab:T1538I; ORF1ab:T3255I; ORF1ab:Q3729R; ORF1ab:P4715L; ORF1ab:P5743S;  S:T95I; S:Y144S; S:R346K; S:E484K; S:N501Y; S:D614G; S:P681H; S:D950N; ORF3a:Q57H; ORF8:T11K; ORF8:P38S; ORF8:S67F; N:T205I
and the following B.1.623 in https://outbreak.info/situation-reports?pango=B.1.623
ORF1ab:P4715L; S:N501Y; S:D614G; S:P681H; ORF3a:Q57H; with 9 AA changes missing

Attached please find  a fasta with 10 of the sequences. 
I use:
pangolin==3.0.3
lineages-2020-05-19-2
pangoLEARN-2021-06-15
scorpio-0.3
constellations-0.0.5
pango-designation v1.2.13
All the best and thanks again fro the great tools you provide,

Sorry, just checked on GISAID (17/06/2021, 1:24:19 PM), and one of the closest sequences according to USHER, EPI_ISL_2220456,  is also classified as B.1.623 (version 3.1.1, lineages version 2021-06-15), although the mutations eg: 
Spike D614G, Spike D950N, Spike E484K, Spike ins145N, Spike N501Y, Spike P681H, Spike R346K, Spike T95I, Spike Y144T, Spike Y145S
would rather fit B.1.621 https://github.com/cov-lineages/pango-designation/issues/62
than B.1.623 https://github.com/cov-lineages/pango-designation/issues/25
Lukas

_________________________________
[b.1.623_run_994_10.fasta.gz](https://github.com/cov-lineages/pangolin/files/6669634/b.1.623_run_994_10.fasta.gz)
Dr. Lukas Endler
Senior Postdoc
_________________________________
CeMM 
Research Center for Molecular Medicine
of the Austrian Academy of Sciences
Lazarettgasse 14, AKH BT 25.3
1090 Vienna, Austria 
Phone +43-1/40160-70055
lendler@cemm.oeaw.ac.at
www.cemm.at

**Comment by rmcolq** [comment]
Hi @luenling - I think this is a designation issue so I have copied it over there and am closing it here. I suspect we need to add some more sequences to the designation list to capture more of the diversity now present in B.1.621.

**Comment by luenling** [comment]
Dear Rachel @rmcolq, 
I am not sure about that. As the two lineages are quite different in their mutations, I thought that was an issue of Pangolin - maybe something went wrong with the training set. Also Usher identified the sequences as B.1.621

All the best and thanks again,
Lukas

**Comment by rmcolq** [comment]
Hi Lukas! 

With your set of 10 sequences, when I run using the default pangoLEARN model it assigns B.1.623, but when I run pangolin using `pangolin --usher` it assigns B.1.621. As you say, when they are placed on a mutation annotated tree by usher, it is clear that they are B.1.621. 

The main reason for misassignments by the (default) pangoLEARN model are down to the input training data. We build a model based on pango-designations (https://github.com/cov-lineages/pango-designation) and this model is very sensitive to the input data, as that determines what the structure of the decision tree will be. This training is unsupervised and selects the sites which are thought to be most informative of a lineage - the model does not consider all the lineage defining mutations for every lineage. The accuracy of pangoLEARN therefore declines as it encounters sequences more unlike those it has been trained on (especially now that mutations commonly occur many times independently), and the best solution to misassignment by pangoLEARN is to provide the newer sequences to the model via a pango-designations issue so that we can add that new diversity into the model.

Alternatively, you are welcome to continue to use pangolin in `--usher` mode which handles new diversity better.
